aeron-archive/src/test/java/io/aeron/archive/CatalogTest.java

```diff
@@ -847,19 +931,362 @@ void extendRecordingShouldUpdateChecksum()
+            assertChecksum(catalog, recordingTwoId, 224, -1059517110, checksum);
+
+            verifyRecordingForId(
+                catalog,
+                recordingThreeId,
+                352,
+                0L,
+                0L,
+                0L,
+                1L,
+                0, SEGMENT_LENGTH, TERM_LENGTH, MTU_LENGTH, 8,
+                3,
+                "channelThatIsVeryLongAndShouldNotBeTruncated",
+                "channelThatIsVeryLongAndShouldNotBeTruncated?tag=f",
+                "source can also be a very very very long String and it will not be truncated even if gets " +
+                "very very long");
+            assertChecksum(catalog, recordingThreeId, 352, -529628341, checksum);
+        }
+    }
+
+    @Test
+    void replaceRecordingShouldUpdateMetadataInPlaceWhenShorterThanTheOldOne()
+    {
+        final long updateTime = 555555L;
+        final Checksum checksum = crc32();
+        try (Catalog catalog =
+            new Catalog(archiveDir, null, 0, CAPACITY, () -> updateTime, checksum, segmentFileBuffer))
+        {
+            catalog.startPosition(recordingOneId, 0);
+            catalog.startPosition(recordingTwoId, 0);
+            catalog.startPosition(recordingThreeId, 0);
+            assertChecksum(catalog, recordingOneId, 160, 729050496, checksum);
+            assertChecksum(catalog, recordingTwoId, 160, 1825380178, checksum);
+            assertChecksum(catalog, recordingThreeId, 352, -1553583482, checksum);
+        }
+
+        try (Catalog catalog =
+            new Catalog(archiveDir, null, 0, CAPACITY, clock, checksum, segmentFileBuffer))
+        {
+            final long originalCapacity = catalog.capacity();
+
+            catalog.replaceRecording(
+                recordingOneId,
+                128,
+                512,
+                111L,
+                222L,
+                -19091,
+                SEGMENT_LENGTH * 16,
+                TERM_LENGTH * 4,
+                1372,
+                21,
+                8,
+                "A",
+                "B",
+                "C");
+
+            assertEquals(recordingThreeId + 1, catalog.nextRecordingId());
+            assertEquals(originalCapacity, catalog.capacity());
+            final int oldFrameLength = 160;
+            verifyRecordingForId(
+                catalog,
+                recordingOneId,
+                oldFrameLength,
+                128,
+                512,
+                111L,
+                222L,
+                -19091,
+                SEGMENT_LENGTH * 16,
+                TERM_LENGTH * 4,
+                1372,
+                21,
+                8,
+                "A",
+                "B",
+                "C");
+            final int newFrameLength = 96;
+            assertChecksum(catalog, recordingOneId, newFrameLength, 1488471744, checksum);
+            // verify that the excessive old data was erased
```

Comment:
Best to pull out as a named method so use of comment is not needed.

Reply:
Done.